### PR TITLE
Fix wrong fall damage when switching gamemodes

### PR DESCRIPTION
**Pull request by XuZhen86**
```This is my first ever pull request, please have mercy if I did something wrong😄```

This pull request tries to fix bug that when switching to game mode 0 from game mode 1, the player receives incorrect fall damage.

Steps to recreate:
1. In game mode 0, land on floor
2. Switch to game mode 1 and fly to a certain height
3. Lower the height to barely touching the floor (not actually standing on the floor)
4. Switch back to game mode 0 and the player receives fall damage. The damage can sometimes kill the player. 

Observed behavior:
The player receives fall damage calculated from the maximum height during game mode 1

Expected behavior:
The player receives fall damage calculated from the location where the player's game mode changes. For example, the player should not receive damage when he/she's only 2 blocks above the floor when the game mode switches. 

Solution:
I have added an extra statement to reset the player's fall distance when switching game mode. At line 75 of file ```GameModeCommand.java```

### CODE REVIEW

File: src/main/java/net/glowstone/command/minecraft/GameModeCommand.java [hunk 1]
```diff
@@ -73,22 +72,20 @@ public boolean execute(CommandSender sender, String label, String[] args) {
     private void updateGameMode(CommandSender sender, Player who, GameMode gameMode) {
         String gameModeName = GameModeUtils.prettyPrint(gameMode);
         who.setGameMode(gameMode);
+        who.setFallDistance(0);
```
Comment (aramperes) [inline src/main/java/net/glowstone/command/minecraft/GameModeCommand.java]: Maybe add this statement to the `GlowPlayer#setGameMode` method instead? That way, this fix would also apply to plugins changing a player's game mode.

**Comment by mastercoms** [comment]
Hi, thank you for your PR to Glowstone! We love to see new contributors!

Since this is your first time here, you may want to check out our [code style](https://github.com/GlowstoneMC/Glowstone/wiki/Code-Style).

File: src/main/java/net/glowstone/command/minecraft/GameModeCommand.java [hunk 2]
```diff
@@ -22,8 +22,7 @@
      * Creates the instance for this command.
      */
     public GameModeCommand() {
-        super("gamemode", "Change the game mode of a player.", "/gamemode <mode> [player]",
-            Collections.emptyList());
+        super("gamemode", "Change the game mode of a player.", "/gamemode <mode> [player]", Collections.emptyList());
```
Comment (aramperes) [inline src/main/java/net/glowstone/command/minecraft/GameModeCommand.java]: As per our code style, the line length should not exceed 100. Can you revert this change?

File: src/main/java/net/glowstone/command/minecraft/GameModeCommand.java [hunk 3]
```diff
@@ -74,21 +73,18 @@ private void updateGameMode(CommandSender sender, Player who, GameMode gameMode)
         String gameModeName = GameModeUtils.prettyPrint(gameMode);
         who.setGameMode(gameMode);
         if (!sender.equals(who)) {
-            sender.sendMessage(
-                who.getDisplayName() + "'s game mode has been updated to " + ChatColor.GRAY + ""
+            sender.sendMessage(who.getDisplayName() + "'s game mode has been updated to " + ChatColor.GRAY + ""
```
Comment (aramperes) [inline src/main/java/net/glowstone/command/minecraft/GameModeCommand.java]: Revert this change as well

File: src/main/java/net/glowstone/command/minecraft/GameModeCommand.java [hunk 4]
```diff
@@ -74,21 +73,18 @@ private void updateGameMode(CommandSender sender, Player who, GameMode gameMode)
...
                     + ChatColor.ITALIC + gameModeName + " Mode" + ChatColor.RESET);
         }
-        who.sendMessage(
-            "Your game mode has been updated to " + ChatColor.GRAY + "" + ChatColor.ITALIC
-                + gameModeName + " Mode" + ChatColor.RESET);
+        who.sendMessage("Your game mode has been updated to " + ChatColor.GRAY + "" + ChatColor.ITALIC + gameModeName
```
Comment (aramperes) [inline src/main/java/net/glowstone/command/minecraft/GameModeCommand.java]: Revert this change as well

File: src/main/java/net/glowstone/command/minecraft/GameModeCommand.java [hunk 5]
```diff
@@ -74,21 +73,18 @@ private void updateGameMode(CommandSender sender, Player who, GameMode gameMode)
...
+                + " Mode" + ChatColor.RESET);
     }
 
     @Override
-    public List<String> tabComplete(CommandSender sender, String alias, String[] args)
-        throws IllegalArgumentException {
+    public List<String> tabComplete(CommandSender sender, String alias, String[] args) throws IllegalArgumentException {
```
Comment (aramperes) [inline src/main/java/net/glowstone/command/minecraft/GameModeCommand.java]: Revert this change as well

**Review by aramperes — CHANGES_REQUESTED** [review]

**Review by aramperes — APPROVED** [review]

**Review by mastercoms — APPROVED** [review]

**Comment by mastercoms** [comment]
Nice work! Thank you for your contribution to Glowstone.
